provider/aws/aws.go

```diff
@@ -140,80 +149,46 @@ func (p *AWSProvider) Ensure(configs map[provider.Resource]map[string]*net.IPNet
 	return nil
 }
 
-func configsToTags(configs map[provider.Resource]map[string]*net.IPNet) map[string]string {
-	tags := make(map[string]string, len(configs))
-	for config, ipAddresses := range configs {
-		addresses := make([]string, 0, len(ipAddresses))
-		for address := range ipAddresses {
-			addresses = append(addresses, address)
-		}
-		sort.Strings(addresses)
-		tags[egressConfigTagPrefix+"configmap/"+config.Namespace+"/"+config.Name] = strings.Join(addresses, ",")
+func stringSetEqual(a, b map[string]struct{}) bool {
+	if len(a) != len(b) {
+		return false
 	}
-	return tags
-}
 
-func parseTagsToEgressConfigStore(tags []*cloudformation.Tag) map[provider.Resource]map[string]*net.IPNet {
-	store := make(map[provider.Resource]map[string]*net.IPNet)
-	for _, tag := range tags {
-		cfg := parseTagAsEgressConfig(tag)
-		if cfg != nil {
-			store[cfg.Resource] = cfg.IPAddresses
+	for ak := range a {
+		if _, ok := b[ak]; !ok {
+			return false
 		}
 	}
-	return store
+	return true
 }
 
-// parseTagAsEgressConfig parses tags of the format
-// egress-config/<resource-type>/<namespace>/<name>=<ipAddress>,
-func parseTagAsEgressConfig(tag *cloudformation.Tag) *provider.EgressConfig {
-	key := aws.StringValue(tag.Key)
-	value := aws.StringValue(tag.Value)
-	if strings.HasPrefix(key, egressConfigTagPrefix) {
-		parts := strings.Split(key, "/")
-		if len(parts) != 4 {
-			return nil
-		}
-		ips := strings.Split(value, ",")
-		ipsSet := make(map[string]*net.IPNet)
-		for _, ip := range ips {
-			_, ipnet, err := net.ParseCIDR(ip)
-			if err != nil {
-				continue
-			}
-			ipsSet[ipnet.String()] = ipnet
-		}
-		return &provider.EgressConfig{
-			Resource: provider.Resource{
-				Name:      parts[3],
-				Namespace: parts[2],
-			},
-			IPAddresses: ipsSet,
-		}
+// parses CIDRs from the Cloudformation template.
+func getCIDRsFromTemplate(template string) (map[string]struct{}, error) {
+	var cfTemplate struct {
+		Resources map[string]interface{}
```

Comment:
I wonder if we could create a better type to not having to to cast empty interfaces.
You can define it sparse and only the data you need rest will be skipped.

Reply:
Good point. I kind of expected it to fail when some resources wouldn't match on the `Properties`, but that does indeed work. I have changed it.